gateway/policies/regex-guardrail/v0.1.0/regexguardrail.go

```diff
@@ -0,0 +1,227 @@
+package regexguardrail
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	policy "github.com/wso2/api-platform/sdk/gateway/policy/v1alpha"
+	utils "github.com/wso2/api-platform/sdk/utils"
+)
+
+const (
+	GuardrailErrorCode         = 446
+	GuardrailAPIMExceptionCode = 900514
+)
+
+// RegexGuardrailPolicy implements regex-based content validation
+type RegexGuardrailPolicy struct{}
+
+// NewPolicy creates a new RegexGuardrailPolicy instance
+func NewPolicy(
+	metadata policy.PolicyMetadata,
+	initParams map[string]interface{},
+	params map[string]interface{},
+) (policy.Policy, error) {
+	return &RegexGuardrailPolicy{}, nil
+}
+
+// Mode returns the processing mode for this policy
+func (p *RegexGuardrailPolicy) Mode() policy.ProcessingMode {
+	return policy.ProcessingMode{
+		RequestHeaderMode:  policy.HeaderModeSkip,
+		RequestBodyMode:    policy.BodyModeBuffer, // Need full body for validation
+		ResponseHeaderMode: policy.HeaderModeSkip,
+		ResponseBodyMode:   policy.BodyModeBuffer, // Need full body for validation
+	}
+}
+
+// OnRequest validates request body against regex pattern
+func (p *RegexGuardrailPolicy) OnRequest(ctx *policy.RequestContext, params map[string]interface{}) policy.RequestAction {
+	// Extract request-specific parameters
+	var requestParams map[string]interface{}
+	if reqParams, ok := params["request"].(map[string]interface{}); ok {
+		requestParams = reqParams
+	} else {
+		return policy.UpstreamRequestModifications{}
+	}
+
+	// Validate parameters
+	if err := p.validateParams(requestParams); err != nil {
+		return p.buildErrorResponse("Parameter validation failed", err, false, false).(policy.RequestAction)
+	}
+
+	var content []byte
+	if ctx.Body != nil {
+		content = ctx.Body.Content
+	}
+	return p.validatePayload(content, requestParams, false).(policy.RequestAction)
+}
+
+// OnResponse validates response body against regex pattern
+func (p *RegexGuardrailPolicy) OnResponse(ctx *policy.ResponseContext, params map[string]interface{}) policy.ResponseAction {
+	// Extract response-specific parameters
+	var responseParams map[string]interface{}
+	if respParams, ok := params["response"].(map[string]interface{}); ok {
+		responseParams = respParams
+	} else {
+		return policy.UpstreamResponseModifications{}
+	}
+
+	// Validate parameters
+	if err := p.validateParams(responseParams); err != nil {
+		return p.buildErrorResponse("Parameter validation failed", err, true, false).(policy.ResponseAction)
+	}
+
+	var content []byte
+	if ctx.ResponseBody != nil {
+		content = ctx.ResponseBody.Content
+	}
+	return p.validatePayload(content, responseParams, true).(policy.ResponseAction)
+}
+
+// validatePayload validates payload against regex pattern
+func (p *RegexGuardrailPolicy) validatePayload(payload []byte, params map[string]interface{}, isResponse bool) interface{} {
+	regexPattern, _ := params["regex"].(string)
```

Comment:
Shouldn't we check `nil` before casting? If `params["regex"]` becomes `nil` then this would become a NPE or panic expression